lib/kubernetestoken/token_validator.go

```diff
@@ -67,8 +74,8 @@ func (v *Validator) getClient() (kubernetes.Interface, error) {
 	return client, nil
 }
 
-// Validate uses the Kubernetes TokenReview API to validate a token and return its UserInfo
-func (v *Validator) Validate(ctx context.Context, token string) (*v1.UserInfo, error) {
+// Validate uses the Kubernetes TokenReview API to validate a name and return its UserInfo
+func (v *TokenReviewValidator) Validate(ctx context.Context, token string) (*ServiceAccountClaims, error) {
```

Comment:
Curious on thoughts on the TokenReview based validator returning the parsed JWT rather than the information returned by the TokenReview endpoint. It makes the implementation in `lib/auth/join_kubernetes.go` simpler as then both `in_cluster` and `static_jwks` implementations return the same type.